content/en/development/design-docs/load-balancer-admin-state-design.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Proactive Traffic Redirection with Azure Load Balancer Admin State"
+linkTitle: "Load Balancer Admin State"
+type: docs
+description: >
+    Use the Azure Load Balancer administrative state to drain nodes immediately during maintenance or spot VM evictions.
+---
+
+## Overview
+
+The cloud-provider-azure component will leverage the Azure Load Balancer (ALB) administrative state feature to instantly stop routing new connections to nodes that are draining, evicted, or otherwise leaving service. When the provider detects either (a) the addition of the GA `node.kubernetes.io/out-of-service` taint or (b) a `PreemptScheduled` event signaling an imminent Spot eviction, it immediately marks that node's backend entry as `Down` on every managed load balancer. The latter path is persisted by adding the taint `cloudprovider.azure.microsoft.com/draining=spot-eviction` so the signal survives controller restarts. This bypasses ALB's probe-based detection window (typically two failed probes at a five-second interval) and reduces the time unhealthy nodes receive traffic by up to ten seconds. Because the change is transparent to the Kubernetes API surface and applies uniformly to Standard Load Balancer deployments, dual-stack services, and both NIC- and IP-based backend pools, every Kubernetes cluster using cloud-provider-azure benefits without any configuration updates. If the out-of-service or draining taint is later removed, the provider restores the admin state to `None` so traffic resumes normally.
+
+## Motivation
+
+- **Eliminate probe lag:** Today ALB may continue to forward traffic to a draining node until health probes fail, extending the window of client errors and timeouts.
+- **Protect upgrades and spot workloads:** During planned maintenance and spot VM evictions, multiple services may suffer cascaded failures because traffic still lands on nodes that are already draining.
+- **Preserve platform simplicity:** Customers should not have to modify Service manifests, CLI invocations, or policies to get better drain behavior. The feature must “just work.”
+
+## Goals
+
+- Cut over traffic away from draining or evicted nodes immediately by setting ALB admin state to `Down`.
+- Apply uniformly to all load-balanced Kubernetes services regardless of `externalTrafficPolicy` or backend pool type.
+- Restore admin state to `None` when a node returns to service so steady-state behavior is unaffected.
+- Avoid additional user-facing APIs, annotations, or CLI switches.
+
+## Non-Goals
```

Comment:
It looks like a non-goal is optimizing LB backend pool membership when an underlying node VM goes offline (hard down)? I assume the existing Azure LB backend pool polling (LB regularly polls backend pool IP addresses) is sufficient to automatically evict downed VMs from the backend pool?